extensions/default/src/ViewerLayout/index.tsx

```diff
@@ -50,13 +50,16 @@ function ViewerLayout({
     //   dataSourceIdx === -1
     //     ? undefined
     //     : `datasources=${pathname.substring(dataSourceIdx + 1)}`;
-
+    
     // Todo: Handle parameters in a better way.
     const query = new URLSearchParams(window.location.search);
     const configUrl = query.get('configUrl');
 
+    const dataSourceName = pathname.substring(dataSourceIdx + 1);
+    const existingDataSource = extensionManager.getDataSources(dataSourceName);
```

Comment:
Are we anticipating a data source to be in the URL that is NOT an actual, configured data source? If so, under what circumstances would this occur? I am a bit confused as to why we need this check.

Reply:
I might have a mode that does not use `/${mode.routeName}/${dataSourceName}`. This means that my mode can rely on the defaultDataSource from the viewer config to define its dataSource. This way I don't need to add the datasources query param and change the defaultDataSource of the worklist coming from a random mode URL e.g. /myModeURL/otherParamWhichIsNotNecessarilyADataSourceName (no dataSource being used)

Reply:
Ok cool. Thanks.